core/src/main/java/org/mskcc/cbio/portal/scripts/NormalizeExpressionLevels.java

```diff
@@ -558,73 +577,79 @@ public static boolean isNormal(String name) {
     # The different sample types for (4) are:
     #  ...
     # 11  normal tissue (not always matched to a cancer sample, used for mRNA, microRNA, methylation) 
-   */
-   public static boolean isTCGANormal(String name){
-      String suffix = "";
-      int dash = 0;
-      search:
-      for(int i=0;i<name.length();i++){
-         if(name.charAt(i)=='-')
-            dash++;
-         if(dash == 3){
-            suffix = name.substring(i,name.length());
-            break search;
-         }
-      }
-      if(suffix.indexOf(TCGA_NORMAL_SUFFIX) == 0 )
-         return true;
-      return false;
-   }
-   
-   
-   private static double avg(double[]v){
-      double avg = 0;
-      for(int i=0;i<v.length;i++)
-         avg=avg+v[i];
-      avg=avg/(double)v.length;
-      return avg;
-   }
-   
-   private static double std(double[]v,double avg){
-      double std = 0;
-      for(int i=0;i<v.length;i++)
-         std=std+Math.pow((v[i]-avg),2);
-      std=std/(double)(v.length-1);
-      std=Math.sqrt(std);
-      return std;
-   }
-   
-   private static double[] resize(double[]v, int s){
-      double[]tmp = new double[s];
-      for(int i=0;i<s;i++)
-         tmp[i]=v[i];
-      return tmp;
-   }
-   
-   private static Integer findColumnIndex(String[] values, String column) {
-       for (int i=0;i<values.length;i++) {
-           if (values[i].equalsIgnoreCase(column)) {
-               return i;
-           }
-       }
-       return -1;
-   }
+    */
+    public static boolean isTCGANormal(String name) {
+        String suffix = "";
+        int dash = 0;
+        search:
+        for (int i=0;i<name.length();i++) {
+            if (name.charAt(i)=='-') {
+                dash++;
+            }
+            if (dash == 3) {
+                suffix = name.substring(i,name.length());
+                break search;
+            }
+        }
+        return (suffix.indexOf(TCGA_NORMAL_SUFFIX) == 0 );
+    }
 
-   private static void fatalError(String msg){
-      throw new RuntimeException("NormalizeExpressionLevels: Fatal error: " + msg );
-   }
    
-   private static void warning(String msg){
-      System.err.println( "NormalizeExpressionLevels: " + msg );
-   }
+    private static double avg(double[]v) {
+        double avg = 0;
+        for (int i=0;i<v.length;i++) {
```

Comment:
There are now standard library functions for some of these (avg at least). See:
java.util.DoubleSummaryStatistics
or guava's com.google.common.math.Stats

It seems unwise for us to be writing basic math functions like this. On the other hand, this PR does not need to refactor the existing code.

Reply:
@sheridancbio I didn't actually want to make changes to the code itself that didn't involve resolving the canonical gene from entrez gene id or hugo symbol. All other changes are strictly ws changes/styling